core/src/boot.py

```diff
@@ -3,14 +3,43 @@
 import storage.sd_salt
 from trezor import config, io, log, loop, res, ui, utils
 from trezor.pin import pin_to_int, show_pin_timeout
+from trezor.ui.confirm import HoldToConfirm
+from trezor.ui.text import Text
 
 from apps.common.request_pin import PinCancelled, request_pin
 from apps.common.sd_salt import SdProtectCancelled, request_sd_salt
 
 
+async def assert_sdcard() -> None:
+    sd = io.SDCard()
+    if not sd.present():
+        return
+
+    sd.power(True)
+    fs = io.FatFS()
+    try:
+        fs.mount()
+    except OSError:
+        text = Text("SD card", ui.ICON_WRONG, ui.RED)
+        text.normal("Unrecognized filesystem")
+        text.normal("found. Unplug the device")
+        text.normal("and remove the card or")
+        text.normal("you can format the card")
+        text.normal("losing all the data on it.")
+        dialog = HoldToConfirm(text, confirm="Format SD card")
+        await dialog
```

Comment:
How about we put this into a local function e.g. `prompt_formatting` or similar?

```
async def assert_sdcard() -> None:
   def prompt():
      text = ...
```

Reply:
The SD card dialogs are in apps/common/sd_salt.py, so that's where this should go too. It's called sd_salt, but the dialogs are generic and apply to any SD card operation, so maybe we shoud rename to sd_card.py.